src/doc/trpl/closures.md

```diff
@@ -181,6 +174,39 @@ fn main() {
 
 Doing this is not particularly common, but it's useful every once in a while.
 
+Before we move on, let us look at a function that accepts two closures.
+
+```{rust}
+fn compose<F, G>(x: i32, f: F, g: G) -> i32
+    where F: Fn(i32) -> i32, G: Fn(i32) -> i32 {
+    g(f(x))
+}
+
+fn main() {
+    compose(5,
+            |&: n: i32| { n + 42 },
+            |&: n: i32| { n * 2 }); // evaluates to 94
+}
+```
+
+You might ask yourself: why do we need to introduce two type
+parameters `F` and `G` here?  Evidently, both `f` and `g` have the
+same signature: `Fn(i32) -> i32`.
+
+That is because in Rust each closure has its own unique type.
+So, not only do closures with different signatures have different types,
+but different closures with the *same* signature have *different*
+types, as well!
+You can think of it this way: the behaviour of a closure is part of its type.
```

Comment:
If this was intended to be a separate paragraph (and it should be), you need another newline here.